Formula/purescript.rb

```diff
@@ -14,21 +14,17 @@ class Purescript < Formula
 
   depends_on "ghc" => :build
   depends_on "haskell-stack" => :build
+  depends_on macos: :catalina
```

Comment:
It really only works on catalina?

Reply:
`ghc` no longer works on Mojave, and @fishtreesugar tried building with `ghc@8.8` in e4abd97f9b6ec01896402b49f2fda3898998da93.

Reply:
You could _maybe_ get it to build on Mojave by letting `stack` pull in its own build GHC rather than using the `ghc` formula.

Reply:
Why not Big Sur though?

Reply:
Ah, `depend_on macos: :catalina` is misleading. It actually means Catalina or newer...

Reply:
Ok, we can get rid of this now/shortly. I've triggered a merge on #78989. You might want to add a build dep on `llvm` for ARM to get this to build on ARM too.

Adding `depends_on "llvm" => :build` everywhere isn't great, but I'm not quite sure of the right way to fix this yet.